clip_retrieval/clip_client.py

```diff
@@ -0,0 +1,209 @@
+"""Clip client is a simple python module that allows you to query the backend remotely."""
+
+import base64
+import enum
+import json
+from pathlib import Path
+from typing import Dict, List, Union
+
+import requests
+from tqdm import tqdm
+
+
+class Modality(enum.Enum):
+    IMAGE = "image"
+    TEXT = "text"
+
+
+class Result():
+    def __init__(self, data_id: int, image_url: str, caption: float, similarity: float):
+        self.data_id = data_id
+        self.image_url = image_url
+        self.caption = caption
+        self.similarity = similarity
+
+
+def get_results(response: List[Dict]) -> List[Result]:
+    results = []
+    for result in response:
+        results.append(Result(data_id=result["id"], image_url=result["image_url"],
+                       caption=result["caption"], similarity=result["similarity"]))
+    return results
+
+
+class ClipClient:
+    """Remotely query the CLIP backend via REST"""
+
+    def __init__(
+        self,
+        url: str,
+        result_dir: str = "results",
```

Comment:
What is this used for ?
I think it might make more sense to let the user store the results only if needed. What do you think?

Reply:
Ah yeah I agree, it makes more sense to keep that simpler.